cadc-vosi/src/main/java/ca/nrc/cadc/vosi/CapGetAction.java

```diff
@@ -0,0 +1,202 @@
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.vosi;
+
+import ca.nrc.cadc.net.HttpTransfer;
+import ca.nrc.cadc.reg.AccessURL;
+import ca.nrc.cadc.reg.Capabilities;
+import ca.nrc.cadc.reg.CapabilitiesReader;
+import ca.nrc.cadc.reg.CapabilitiesWriter;
+import ca.nrc.cadc.reg.Capability;
+import ca.nrc.cadc.reg.Interface;
+import ca.nrc.cadc.reg.client.LocalAuthority;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import ca.nrc.cadc.rest.InlineContentHandler;
+import ca.nrc.cadc.rest.RestAction;
+import ca.nrc.cadc.rest.SyncOutput;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author pdowler
+ */
+public class CapGetAction extends RestAction {
+    private static final Logger log = Logger.getLogger(CapGetAction.class);
+
+    /**
+     * Enable transformation of the capabilities template (default: true). Subclasses
+     * may disable this according to some policy. The current transform is to change
+     * the host name in every accessURL in the capabilities to match the host name used
+     * in the request. This works fine in most cases but would not work
+     * if some accessURL(s) within an application are deployed on a different host.
+     * For example, if the VOSI-availability endpoint is deployed on an separate host
+     * so it can probe the service from the outside, then capabilities transform
+     * would need to be disabled.
+     */
+    protected boolean doTransform = true;
+    
+    /**
+     * Enable injection of additional capabilities for external authentication
+     * providers (default: true).
+     */
+    protected boolean injectAuthProvider = false;
+    
+    public CapGetAction() {
+        super();
+    }
+
+    @Override
+    protected InlineContentHandler getInlineContentHandler() {
+        return null;
+    }
+
+    @Override
+    public void doAction() throws Exception {
+        Capabilities caps = CapInitAction.getTemplate(appName, componentID);
+
```

Comment:
I guess because InitAction does not inherit from RestAction is does not have appName and componentID.    Seems a bit strange for another action to be telling InitAction its own context here.

Reply:
both kinds of actions get the same appName and componentID; the purpose of this static method is to hide the way CapInitAction stores the template in JNDI.. but the static method doesn't have access to the values of those things. I will probably just remove appName from there since it isn't needed and it's just an agreement between the two actions.